var/spack/repos/builtin/packages/pgplot/package.py

```diff
@@ -99,6 +99,7 @@ def edit(self, spec, prefix):
             enable_driver('! PNDRIV 1 /PNG')
             sub['@CCOMPL@'] += ' -I{0}'.format(self.spec['libpng'].prefix.include)
             sub['@FFLAGD@'] += ' -L{0} -lpng'.format(self.spec['libpng'].prefix.lib)
+            sub['@FFLAGD@'] += ' -L{0} -lz'.format(self.spec['zlib'].prefix.lib)
```

Comment:
`pgplot` does not use zlib. `libpng` dependes on `zlib` so I have to explicitly tell makefile to include it. The compiler wrapper by spack should have taken care of it but it is not.

Reply:
It would be better to write
```Python
sub['@CCOMPL@'] += ' {0}'.format(self.spec['libpng'].headers.cpp_flags)
sub['@FFLAGD@'] += ' {0}'.format(self.spec['libpng'].libs.ld_flags)
```
if that works. I hope that `libpng`'s dependencies are included here. If so, the `depends_on('zlib')` can also be removed.

Reply:
I think it is not working if I change to it to this:

```diff
         if '+png' in spec:
             enable_driver('! PNDRIV 1 /PNG')
-            sub['@CCOMPL@'] += ' -I{0}'.format(self.spec['libpng'].prefix.include)
-            sub['@FFLAGD@'] += ' -L{0} -lpng'.format(self.spec['libpng'].prefix.lib)
-            sub['@FFLAGD@'] += ' -L{0} -lz'.format(self.spec['zlib'].prefix.lib)
+            sub['@CCOMPL@'] += ' {0}'.format(self.spec['libpng'].headers.cpp_flags)
+            sub['@FFLAGD@'] += ' {0}'.format(self.spec['libpng'].libs.ld_flags)

             filter_file('pndriv.o : ./png.h ./pngconf.h ./zlib.h ./zconf.h',
-                        'pndriv.o : {0}/png.h {0}/pngconf.h {1}/zlib.h {1}/zconf.h'.
-                        format(self.spec['libpng'].prefix.include,
-                               self.spec['zlib'].prefix.include), 'makemake')
+                        'pndriv.o :', 'makemake')
+
             self.rpath.append(self.spec['libpng'].prefix)
```

And the replaced `g77_gcc.conf` file contains no information of zlib like this:
```   
   FFLAGD="-nofor-main -L/work/packages/spack_20200621/opt/spack/linux-centos7-westmere/intel-2021.2.0/libpng-1.6.37-sr7zyzuno7kv2wnh5kqjycgs62o7puze/lib -lpng"

# Mandatory.
# The C compiler to use.

   CCOMPL="/work/packages/spack_20200621/opt/spack/linux-centos7-nehalem/gcc-4.8.5/intel-oneapi-compilers-2021.2.0-muplh6xossdrscjp7uzrjssunymlvinr/compiler/2021.2.0/linux/bin/intel64/icc -I/work/packages/spack_20200621/opt/spack/linux-centos7-westmere/intel-2021.2.0/libpng-1.6.37-sr7zyzuno7kv2wnh5kqjycgs62o7puze/include"
```

Reply:
Thanks for trying.